apps/emqx/src/emqx_persistent_session_ds_state.erl

```diff
@@ -359,6 +367,12 @@ del_rank(Key, Rec) ->
 fold_ranks(Fun, Acc, Rec) ->
     gen_fold(ranks, Fun, Acc, Rec).
 
+-spec session_count() -> non_neg_integer().
+session_count() ->
+    %% N.B.: this is potentially costly.  Should not be called in hot paths.
+    %% `mnesia:table_info(_, size)' is always zero for rocksdb, so we need to traverse...
+    do_session_count(make_session_iterator(), 0).
+
 -spec make_session_iterator() -> session_iterator().
 make_session_iterator() ->
     case mnesia:dirty_first(?session_tab) of
```

Comment:
Q: What this case clause is for?

Reply:
Good question.  I didn't introduce it, but seems like we could "eta reduce" it.  :thinking: